src/main/java/com/medallia/word2vec/Word2VecModel.java

```diff
@@ -35,15 +33,19 @@
 public class Word2VecModel {
 	final List<String> vocab;
 	final int layerSize;
-	final double[] vectors;
+	final DoubleBuffer vectors;
 	private final static long ONE_GB = 1024 * 1024 * 1024;
 
-	Word2VecModel(Iterable<String> vocab, int layerSize, double[] vectors) {
+	Word2VecModel(Iterable<String> vocab, int layerSize, DoubleBuffer vectors) {
 		this.vocab = ImmutableList.copyOf(vocab);
 		this.layerSize = layerSize;
 		this.vectors = vectors;
 	}
```

Comment:
Yep no worries.  If you'd like, you can leave a comment that we avoid a defensive copy for performance.